src/main/java/application/dao/ProductDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package application.dao;
```

Comment:
```suggestion
package application.dao.impl;
```
let's store implementations separate package. Same with with services

Reply:
done